psyneulink/core/components/functions/nonstateful/optimizationfunctions.py

```diff
@@ -31,6 +31,19 @@
 from numbers import Number
 
 import numpy as np
+
+try:
+    import torch
+except ImportError:
+    torch = None
+
+# Conditionally import torch, this is a bit more strict than the usual conditional import of pytorch. This is because
+# GradientOptimization needs torch.func.grad, which is not available in pytorch versions less than 2.0.0.
+try:
+    from torch.func import grad
+except ImportError:
+    grad = None
+
```

Comment:
this can be moved to else of the above try block

Reply:
Yeah, i like the use of `'grad' in dir(torch.func)`. Changed the code to use that.